### ISSUE TITLE
Calico network only works after restarting calico-node

### ISSUE THREAD

**Issue by pennpeng** [apprenda/kismatic#206]
i create a pod，and test the dns server。
```
/ # nslookup  kubernetes.default
nslookup: can't resolve '(null)': Name does not resolve

nslookup: can't resolve 'kubernetes.default': Try again
/ # 
/ # cat /etc/resolv.conf 
search default.svc.cluster.local svc.cluster.local cluster.local
nameserver 172.17.0.2
options ndots:5
/ # 
```
the pod(in node2) ip is 172.16.70.151,  i ping this pod in master node and node1:
```
root@slave01:~# ping 172.16.70.151
PING 172.16.70.151 (172.16.70.151) 56(84) bytes of data.
^C
--- 172.16.70.151 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 1998ms
```
when i restart "docker restart calico-node",it is ok!!!
```
/ # nslookup  my-nginx
nslookup: can't resolve '(null)': Name does not resolve

Name:      my-nginx
Address 1: 172.17.139.142 my-nginx.default.svc.cluster.local
/ # 
/ # 
/ # nslookup  kubernetes.default
nslookup: can't resolve '(null)': Name does not resolve

Name:      kubernetes.default
Address 1: 172.17.0.1 kubernetes.default.svc.cluster.local
```
what's wrong with it ??

**Comment by alexbrand** [comment]
@pennpeng Can you share information about your environment? Where is your cluster running (metal, gcloud, aws)? What version of kismatic are you using? Thanks

**Comment by pennpeng** [comment]
@alexbrand  my  environment is vmware workstation,one master node,two worker node;the version of kismatic is 1.0.3 and 1.1.0 all meet the seem problem。sometimes the ui can't  access，and restart the calico network it is ok。

**Comment by pennpeng** [comment]
today i  run my environment（everyday i poweroff my vm），The problem arose again

**Comment by dkoshkin** [comment]
Do you have DNS configured for the machine hostnames, if not you need `update_hosts_files=true` for the machines to be resolvable via hostnames.

**Comment by alexbrand** [comment]
This sounds like an issue we saw with Calico where the order of how things come up matters. The issue I have in mind is: https://github.com/projectcalico/calicoctl/issues/716. I suspect there is some timing dependencies between `calico-node` and the route configuration being available (the etcd cluster in this case).

If you are able to reproduce, could you post the route table (`ip route`) from a bad node on this issue?

**Comment by pennpeng** [comment]
@alexbrand 
the master
```
[root@master ~]# ping 172.16.60.198
PING 172.16.60.198 (172.16.60.198) 56(84) bytes of data.


^C
--- 172.16.60.198 ping statistics ---
58 packets transmitted, 0 received, 100% packet loss, time 57050ms

[root@master ~]# ip route 
default via 192.168.83.2 dev ens33  proto static  metric 100 
blackhole 172.16.1.64/26  proto bird 
172.16.14.0/26 via 192.168.83.192 dev tunl0  proto bird onlink 
172.16.14.128/26 via 192.168.83.192 dev tunl0  proto bird onlink 
172.16.54.0 dev cali53e077a8dc5  scope link 
blackhole 172.16.54.0/26  proto bird 
172.16.60.192/26 via 192.168.83.191 dev tunl0  proto bird onlink 
172.16.171.192/26 via 192.168.83.191 dev tunl0  proto bird onlink 
172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.0.1 
192.168.83.0/24 dev ens33  proto kernel  scope link  src 192.168.83.190  metric 100 
```
the node2
```
[root@slave02 ~]# ping 172.16.60.198
PING 172.16.60.198 (172.16.60.198) 56(84) bytes of data.


^C
--- 172.16.60.198 ping statistics ---
54 packets transmitted, 0 received, 100% packet loss, time 53041ms

[root@slave02 ~]# ip route 
default via 192.168.83.2 dev ens33  proto static  metric 100 
172.16.1.64/26 via 192.168.83.190 dev tunl0  proto bird onlink 
172.16.14.0 dev calic6f346d4e8c  scope link 
blackhole 172.16.14.0/26  proto bird 
172.16.14.2 dev calif6cc12845bf  scope link 
172.16.14.3 dev cali475be4a81bb  scope link 
blackhole 172.16.14.128/26  proto bird 
172.16.54.0/26 via 192.168.83.190 dev tunl0  proto bird onlink 
172.16.60.192/26 via 192.168.83.191 dev tunl0  proto bird onlink 
172.16.171.192/26 via 192.168.83.191 dev tunl0  proto bird onlink 
172.17.0.0/16 dev docker0  proto kernel  scope link  src 172.17.0.1 
192.168.83.0/24 dev ens33  proto kernel  scope link  src 192.168.83.192  metric 100 
```

**Comment by pennpeng** [comment]
@dkoshkin i have configured dns in my vm
```
[root@master ~]# cat /etc/resolv.conf 
# Generated by NetworkManager
search docoreos.com
nameserver 223.5.5.5
```
and i set update_hosts_files=false```,but  i add the hostfiles in my hand
```
[root@master ~]# cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6

192.168.83.190 master.docoreos.com master
192.168.83.191 slave01.docoreos.com slave01
192.168.83.192 slave02.docoreos.com slave02
```

**Comment by dasmb** [comment]
Calico 2.0 has been merged into master and will be released as part of 1.2 within the next 10 days.
